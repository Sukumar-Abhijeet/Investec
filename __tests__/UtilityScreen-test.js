import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import UtilityScreen from '../src/@Screens/Auth/Utility';

const mockStore = configureMockStore();
const store = mockStore({userData:{userName:"Sukumar Abhijeet"}});

const createTestProps = {
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  }
};

test('Testing Utility Screen .js', () => {
  const snap = renderer.create(
      <Provider store={store}>
        <UtilityScreen  {...createTestProps} />
      </Provider>
  ).toJSON();
  expect(snap).toMatchSnapshot();
});
