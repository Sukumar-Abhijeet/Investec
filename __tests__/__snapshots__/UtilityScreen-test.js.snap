// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Utility Screen .js 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 15.714285714285714,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "#000",
        "fontSize": 22,
        "fontWeight": "bold",
        "margin": 31.428571428571427,
      }
    }
  >
    Welcome - 
    Sukumar Abhijeet
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 31.428571428571427,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25.142857142857142,
          "marginTop": 15.714285714285714,
        }
      }
    >
      Merging The objects
    </Text>
    <Text>
      1st Object : 
      {"test":"case1","value":50}
       
    </Text>
    <Text>
      2nd Object : 
      {"name":"Investec","age":100}
       
    </Text>
    <Text>
      Results : 
      {"test":"case1","value":50,"name":"Investec","age":100}
    </Text>
    <Text
      style={
        Object {
          "fontSize": 25.142857142857142,
          "marginTop": 15.714285714285714,
        }
      }
    >
      Merging The Arrays
    </Text>
    <Text>
      1st Array : 
      [1,2,3,4,5]
       
    </Text>
    <Text>
      2nd Array : 
      ["N1","N2","N3","N4","N5"]
       
    </Text>
    <Text>
      Results : 
      [1,2,3,4,5,"N1","N2","N3","N4","N5"]
    </Text>
    <Text
      style={
        Object {
          "fontSize": 25.142857142857142,
          "marginTop": 15.714285714285714,
        }
      }
    >
      Concatnation Of Strings | Numbers
    </Text>
    <Text>
      1st 
      number
       : 
      500
       
    </Text>
    <Text>
      2nd 
      number
       : 
      300
       
    </Text>
    <Text>
      Results : 
      800
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "bottom": 78.57142857142857,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6eb1f7",
          "borderRadius": 15.714285714285714,
          "marginBottom": 15.714285714285714,
          "opacity": 1,
          "paddingVertical": 23.57142857142857,
          "width": "100%",
        }
      }
      testID="NavigateToButtons"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Visit Buttons
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6eb1f7",
          "borderRadius": 15.714285714285714,
          "marginBottom": 15.714285714285714,
          "opacity": 1,
          "paddingVertical": 23.57142857142857,
          "width": "100%",
        }
      }
      testID="NavigateToNative"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Check Devices
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6eb1f7",
          "borderRadius": 15.714285714285714,
          "marginBottom": 15.714285714285714,
          "opacity": 1,
          "paddingVertical": 23.57142857142857,
          "width": "100%",
        }
      }
      testID="NavigateToLogin"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Check Login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
